server/historian/packages/historian-base/src/services/restGitService.ts

```diff
@@ -246,8 +248,8 @@ export class RestGitService {
             this.getSummaryCacheKey("container"),
             async () => this.get<IWholeFlatSummary>(
                 `/repos/${this.getRepoPath()}/git/summaries/${encodeURIComponent(sha)}`),
-            useCache,
-            true);
+            sha === latestSummarySha ? useCache : false,
```

Comment:
I think that updating `RestGitService.resolve` made it more clear to me... At least, renaming `resolvingSummary` to `resolvingLatestSummary` made more sense to me because even if we adopt the above approach, resolve() would only actually be used by "latest" summaries, so `resolvingSummary` would effectively mean `resolvingLatestSummary`. And I thought that might be confusing if not made explicitly

Reply:
Fair enough :)

Reply:
As we discussed offline, I updated the implementation of both `getSummary()` and `resolve()`. `resolve()` is now simplified, back to what it was before the discovery work, and all the summary-specific logic is centralized in `getSummary()`